src/main/resources/assets/pipez/lang/de_de.json

```diff
@@ -1,45 +1,45 @@
+  "item.pipez.advanced_upgrade": "Fortgeschrittene Rohr Verbesserung",
+  "item.pipez.ultimate_upgrade": "Ultimative Rohr Verbesserung",
+  "item.pipez.infinity_upgrade": "Unendliche Rohr Verbesserung",
   "item.pipez.wrench": "Röhren Zange",
   "item.pipez.filter_destination_tool": "Filter Ziel Gerät",
-  "tooltip.pipez.wrench": "Klicke auf eine Röhre um sie einzustellen",
+  "tooltip.pipez.wrench": "Klicke auf ein Rohr, um sie einzustellen",
   "tooltip.pipez.distribution": "Verteilung: %s",
   "tooltip.pipez.distribution.nearest": "Nächste zuerst",
   "tooltip.pipez.distribution.furthest": "Weiteste zuert",
   "tooltip.pipez.distribution.round_robin": "Rundlauf",
   "tooltip.pipez.distribution.random": "Zufällig",
   "tooltip.pipez.redstone_mode": "Rotstein Modus: %s",
   "tooltip.pipez.redstone_mode.ignored": "Ignoriert",
-  "tooltip.pipez.redstone_mode.off_when_powered": "Aus Wenn Aktiviert",
-  "tooltip.pipez.redstone_mode.on_when_powered": "An Wenn Aktiviert",
+  "tooltip.pipez.redstone_mode.off_when_powered": "Aus wenn Redstone aktiviert",
+  "tooltip.pipez.redstone_mode.on_when_powered": "An wenn Aktiviert",
   "tooltip.pipez.redstone_mode.always_off": "Immer Aus",
   "tooltip.pipez.filter_mode": "Filter Modus: %s",
   "tooltip.pipez.filter_mode.whitelist": "Erlaubt",
   "tooltip.pipez.filter_mode.blacklist": "Nicht Erlaubt",
   "tooltip.pipez.filter.not": "Nicht",
   "tooltip.pipez.filter.accepts_tag": "Alle %s",
-  "tooltip.pipez.filter.item_tag.description": "Tags Präfix mit # oder item IDs",
-  "tooltip.pipez.filter.nbt_string.description": "NBT Data",
-  "tooltip.pipez.filter.nbt_string.no_nbt": "Dieser Typ kan keine NBT tags haben",
-  "tooltip.pipez.filter.nbt.exact": "Genaue übereinstimmung NBT data",
+  "tooltip.pipez.filter.item_tag.description": "Präfix Stichworte mit # oder Item IDs",
+  "tooltip.pipez.filter.nbt_string.description": "NBT Daten",
+  "tooltip.pipez.filter.nbt_string.no_nbt": "Dieser Typ kan keine NBT Stichworte haben",
+  "tooltip.pipez.filter.nbt.exact": "Genaue Übereinstimmung der Stichwort Daten",
   "tooltip.pipez.filter.nbt.not_exact": "Nur bereitgestellte Tags abgleichen ",
   "tooltip.pipez.filter.not_inverted": "Regulär",
   "tooltip.pipez.filter.inverted": "Invertiert",
   "tooltip.pipez.filter_destination_tool": "Klicke auf einen Block, um das Ziel festzulegen",
   "tooltip.pipez.filter_destination_tool.destination": "Ziel: X=%s, Y=%s, Z=%s, %s",
   "tooltip.pipez.filter.destination.description": "Benutze das Filter Ziel Gerät um ein Ziel festzulegen",
   "tooltip.pipez.filter.destination.click_to_remove": "Klicken zum entfernen",
-  "tooltip.pipez.filter.unknown_block": "Unbekannter BLOCK",
+  "tooltip.pipez.filter.unknown_block": "UNBEKANTER BLOCK",
   "tooltip.pipez.filter.destination_location": "Bei X=%s, Y=%s, Z=%s",
   "tooltip.pipez.filter.destination_distance": "%s blöcke(s) entfernt",
   "tooltip.pipez.filter.destination_side": "Zugewandt %s",
```

Comment:
Doesn't really fit

Reply:
Which part do you mean? i fixed the typo aswell.

First of all, im realy sorry for all the misstakes. I realy need to learn to pay more attention and not rush it. 
And i realy realy apprechiate your patience with me.
Im translating with notepad++ would there be an better alternative?

Reply:
"Zugewandt"

No worries :D
VSCode is a good alternative.

Reply:
would "Zielseite" work?

Reply:
what does it "destination_side" stand for? where is it used?

Reply:
It is used for the side of the attached block with the destination tool.

Reply:
what would you recomend? "richtung" ?

Reply:
Yeah, "Richtung" sounds good. The only alternative I had in mind would be "Orientierung"

Reply:
I like "Orientierung" more, i changed it to that. I pushed now the latest changes.